autoscheduler/frontend/src/components/ScheduleCard/ScheduleCard.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import { useSelector } from 'react-redux';
+import * as styles from './ScheduleCard.css';
+import DragHandle from './DragHandle';
+import { RootState } from '../../redux/reducers';
+import { AvailabilityArgs } from '../../types/Availability';
+
+let contentHeight: number = null;
+
+interface BasicProps {
+  startTimeHours: number;
+  startTimeMinutes: number;
+  endTimeHours: number;
+  endTimeMinutes: number;
+  borderColor?: string;
+  backgroundColor?: string;
+  backgroundStripes?: boolean;
+  firstHour: number;
+  lastHour: number;
+  onResizeWindow?: (isBig: boolean) => void;
+  onDragHandleDown?: (evt: React.MouseEvent<HTMLDivElement, MouseEvent>,
+    endSelected: boolean) => void;
+}
+
+type ScheduleCardProps = React.PropsWithChildren<BasicProps>;
```

Comment:
Just for my understanding, but how come you have to use `PropsWithChildren`, rather than just using the props interface directly like you do in other places?

Reply:
`PropsWithChildren` includes `props`, the normal props object that we pass into most components, and `children`, a special argument that contains all of the elements inside of this one. So, for example, if we have:

```
<ScheduleCard>
  <p>Hello</p>
</ScheduleCard>
```
then `children` is the inner `<p>` tag